<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Preferences</class>
 <widget class="QDialog" name="Preferences">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Preferences</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>General</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <layout class="QFormLayout" name="formLayout_2">
         <item row="0" column="0">
          <widget class="QLabel" name="label_2">
           <property name="toolTip">
            <string>Minutes - Seconds (00:00 means no timeout)</string>
           </property>
           <property name="text">
            <string>Request timeout:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QTimeEdit" name="teRequestTimeout">
           <property name="toolTip">
            <string>Minutes - Seconds (00:00 means no timeout)</string>
           </property>
           <property name="displayFormat">
            <string>mm:ss</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QCheckBox" name="cbUseDefaultHeaders">
         <property name="text">
          <string>Use default headers</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="gbRequest">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string>Request</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <layout class="QFormLayout" name="formLayout_3">
            <item row="0" column="0">
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>Request type:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QComboBox" name="cbRequestType">
              <item>
               <property name="text">
                <string>GET</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>POST</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>PUT</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>DELETE</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>PATCH</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>HEAD</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>TRACE</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>OPTIONS</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QGroupBox" name="gbHeaders">
            <property name="title">
             <string>Headers</string>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout">
               <item>
                <widget class="QLabel" name="label_3">
                 <property name="text">
                  <string>Body Type:</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QComboBox" name="cbRequestBodyType">
                 <item>
                  <property name="text">
                   <string>raw</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>form-data</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>x-form-www-urlencoded</string>
                  </property>
                 </item>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_8">
               <item>
                <widget class="QTableWidget" name="twRequestBodyKeyValue">
                 <property name="alternatingRowColors">
                  <bool>true</bool>
                 </property>
                 <column>
                  <property name="text">
                   <string>Key</string>
                  </property>
                 </column>
                 <column>
                  <property name="text">
                   <string>Value</string>
                  </property>
                 </column>
                </widget>
               </item>
               <item>
                <layout class="QVBoxLayout" name="verticalLayout_7">
                 <item>
                  <widget class="QToolButton" name="tbRequestBodyKeyValueAdd">
                   <property name="toolTip">
                    <string>Add new row</string>
                   </property>
                   <property name="text">
                    <string>...</string>
                   </property>
                   <property name="icon">
                    <iconset resource="Resources/resources.qrc">
                     <normaloff>:/icons/plus.png</normaloff>:/icons/plus.png</iconset>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QToolButton" name="tbRequestBodyKeyValueRemove">
                   <property name="toolTip">
                    <string>Remove selected rows</string>
                   </property>
                   <property name="text">
                    <string>...</string>
                   </property>
                   <property name="icon">
                    <iconset resource="Resources/resources.qrc">
                     <normaloff>:/icons/minus.png</normaloff>:/icons/minus.png</iconset>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <layout class="QFormLayout" name="formLayout_4">
         <item row="0" column="0">
          <widget class="QLabel" name="label_8">
           <property name="toolTip">
            <string>Warning: Setting this value too high can slown down or even crash FRequest (since it parses all the received data). 

As alternative try to download the response as file (since this way it writes all the data to a file).

(Default value = 200 KB)</string>
           </property>
           <property name="text">
            <string>Max request response data size to display:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>On startup:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QComboBox" name="cbOnStartup">
           <item>
            <property name="text">
             <string>Ask to load last project</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Load last project</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Do nothing</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_9">
           <property name="text">
            <string>Theme:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QComboBox" name="cbTheme">
           <item>
            <property name="text">
             <string>OS Default</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Jorgen's Dark Theme</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="sbMaxRequestResponseDataSizeToDisplay">
           <property name="toolTip">
            <string>Warning: Setting this value too high can slown down or even crash FRequest (since it parses all the received data). 

As alternative try to download the response as file (since this way it writes all the data to a file).

(Default value = 200 KB)</string>
           </property>
           <property name="buttonSymbols">
            <enum>QAbstractSpinBox::NoButtons</enum>
           </property>
           <property name="suffix">
            <string> KB</string>
           </property>
           <property name="minimum">
            <number>50</number>
           </property>
           <property name="maximum">
            <number>512000</number>
           </property>
           <property name="value">
            <number>200</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QCheckBox" name="cbSaveWindowGeometryWhenExiting">
         <property name="text">
          <string>Save window geometry when exiting</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="cbHideProjectSavedDialog">
         <property name="text">
          <string>Hide confirmation dialog when saving a project</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_3">
      <attribute name="title">
       <string>Network</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <widget class="QCheckBox" name="cbProxyUseProxy">
         <property name="text">
          <string>Use proxy</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="gbProxy">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="title">
          <string>Proxy</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <layout class="QFormLayout" name="formLayout">
            <item row="0" column="0">
             <widget class="QLabel" name="label">
              <property name="text">
               <string>Proxy Type:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QComboBox" name="cbProxyType">
              <item>
               <property name="text">
                <string>Automatic</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Http Transparent Proxy</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Http Proxy</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Socks5 Proxy</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>Hostname:</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QLineEdit" name="leProxyHostname">
              <property name="enabled">
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>Port:</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QLineEdit" name="leProxyPort">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="text">
               <string>3128</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>Projects Authentications Data</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <item>
        <widget class="QLabel" name="lbProjAuthDataNote">
         <property name="text">
          <string>Note:</string>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>FRequest Configuration Projects Authentications</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_8">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QTableWidget" name="twConfigProjAuthData">
              <property name="alternatingRowColors">
               <bool>true</bool>
              </property>
              <property name="sortingEnabled">
               <bool>true</bool>
              </property>
              <column>
               <property name="text">
                <string>Project Last Name</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>Project Uuid</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>Authentication Type</string>
               </property>
              </column>
             </widget>
            </item>
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_10">
              <item>
               <widget class="QToolButton" name="tbConfigProjDataRemove">
                <property name="toolTip">
                 <string>Remove selected rows</string>
                </property>
                <property name="text">
                 <string>...</string>
                </property>
                <property name="icon">
                 <iconset resource="Resources/resources.qrc">
                  <normaloff>:/icons/minus.png</normaloff>:/icons/minus.png</iconset>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="Resources/resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>Preferences</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>Preferences</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
